synapse/handlers/room_list.py

```diff
@@ -288,23 +292,41 @@ def get_order_for_room(room_id):
 
     @defer.inlineCallbacks
     def _append_room_entry_to_chunk(self, room_id, num_joined_users, chunk, limit,
-                                    search_filter):
+                                    search_filter, from_federation=False):
         """Generate the entry for a room in the public room list and append it
         to the `chunk` if it matches the search filter
         """
         if limit and len(chunk) > limit + 1:
             # We've already got enough, so lets just drop it.
             return
 
-        result = yield self.generate_room_entry(room_id, num_joined_users)
+        result = yield self.generate_room_entry(room_id,
+            num_joined_users)
 
-        if result and _matches_room_entry(result, search_filter):
+        if from_federation and not self.config.allow_non_federated_in_public_rooms:
+            if result["m.federate"] == False:
+                # This is a non-federating room and the config has chosen not
+                # to show these rooms to other servers
+                chunk.append(None)
```

Comment:
why do we not just `pass` ?

Reply:
The original code was returning `None` and then appending it to the chunk, so I kept doing so. Not sure if it's necessary or we can just leave it out.

Reply:
Looking at the code, I don't think `None`s in there are particularly useful, so just not inserting anything (by `return`ing instead) seems like the right way to go.